core/modules/wmgr/python/dbMgr.py

```diff
@@ -0,0 +1,692 @@
+    return errorResponse(500, "db.DbException", str(error))
+
+@dbService.route('', methods=['GET'])
+def dbs():
+    """ Return the list of databases """
+
+    _log.debug('request: %s', request)
+    _log.debug('dbs() GET => get database list')
+
+    # use non-privileged account, may limit the list of databases returned.
+    dbConn = Config.instance().dbConn()
+    dbs = dbConn.execCommandN('SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA')
+
+    dbs = [row[0] for row in dbs]
+    _log.debug('dbs(): dbs = %s', dbs)
+
+    dbData = []
+    for dbName in dbs:
+        if dbName not in _specialDbs:
+            dbData += [_dbDict(dbName)]
+
+    return json.jsonify(results=dbData)
+
+
+@dbService.route('', methods=['POST'])
+def dbCreate():
+    """
+    Create new database, database name comes in a query string
+
+    Following parameters are expected to come in a request (in request body
+    with application/x-www-form-urlencoded content like regular form):
+        db: database name (required)
+    """
+
+    _log.debug('request: %s', request)
+    _log.debug('request.form: %s', request.form)
+    _log.debug('dbCreate() POST => make database')
+
+    # get database name from query
+    dbName = request.form.get('db', '').strip()
+    if not dbName:
+        raise ExceptionResponse(400, "MissingArgument", "Database name argument (db) is missing")
+
+    # validate it
+    _validateDbName(dbName)
+
+    # create database, use privileged account
+    dbConn = Config.instance().privDbConn()
+    try:
+        dbConn.createDb(dbName)
+    except db.DbException as exc:
+        _log.error('dbCreate(): exception when creating database %s: %s', dbName, exc)
+        if exc.errCode() == db.DbException.DB_EXISTS:
+            raise ExceptionResponse(409, "DatabaseExists", "Database %s already exists" % dbName)
+        raise
+
+    _log.debug('dbCreate(): database %s created', dbName)
+
+    # grant full access to non-privileged account
+    cmd = "GRANT ALL PRIVILEGES ON {0}.* TO '{1}'@'{2}'"
+    user = Config.instance().dbUser
+    hostnames = ['localhost']
+    for host in hostnames:
+        try:
+            dbConn.execCommand0(cmd.format(dbName, user, host))
+        except db.DbException as exc:
+            _log.error('dbCreate(): exception when adding grants on database: %s', exc)
+            raise ExceptionResponse(500, "GrantFailed", "Database %s created but GRANT failed" % dbName,
+                                    str(exc))
+
+    _log.debug('dbCreate(): grants added')
+
+    # return representation for new database, 201 code is for CREATED
+    response = json.jsonify(result=_dbDict(dbName))
+    response.status_code = 201
+    return response
+
+
+@dbService.route('/<dbName>', methods=['DELETE'])
+def dbDelete(dbName):
+    """ Drop database """
+
+    _log.debug('request: %s', request)
+    _log.debug('request.form: %s', request.form)
+    _log.debug('dbDelete() DELETE => drop database')
```

Comment:
Not sure what kind of confirmation do you mean, this API is not for end-users. And it will be protected by reasonable auth like the rest of the API.